pkg/providers/instancetype/suite_test.go

```diff
@@ -1222,6 +1223,53 @@ var _ = Describe("InstanceType Provider", func() {
 			Expect(kubeletFlags).To(ContainSubstring("--register-with-taints=" + karpv1.UnregisteredNoExecuteTaint.ToString()))
 		})
 	})
+
+	DescribeTable("Azure CNI node labels and agentbaker network plugin", func(
+		networkPlugin, networkPluginMode, networkDataplane, expectedAgentBakerNetPlugin string,
+		expectedNodeLabels sets.Set[string]) {
+		options := test.Options(test.OptionsFields{
+			NetworkPlugin:     lo.ToPtr(networkPlugin),
+			NetworkPluginMode: lo.ToPtr(networkPluginMode),
+			NetworkDataplane:  lo.ToPtr(networkDataplane),
+		})
+		ctx = options.ToContext(ctx)
+
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		pod := coretest.UnschedulablePod()
+		ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+		ExpectScheduled(ctx, env.Client, pod)
+		customData := ExpectDecodedCustomData(azureEnv)
+
+		Expect(customData).To(ContainSubstring(fmt.Sprintf("NETWORK_PLUGIN=%s", expectedAgentBakerNetPlugin)))
+
+		for label := range expectedNodeLabels {
+			Expect(customData).To(ContainSubstring(label))
+		}
+	},
+		Entry("Azure CNI V1",
+			"azure", "", "",
+			"azure", sets.New[string]()),
+		Entry("Azure CNI w Overlay",
+			"azure", "overlay", "",
+			"none",
+			sets.New(
+				"kubernetes.azure.com/azure-cni-overlay=true",
+				"kubernetes.azure.com/network-subnet=karpentersub",
+				"kubernetes.azure.com/nodenetwork-vnetguid=test-vnet-guid",
+				"kubernetes.azure.com/podnetwork-type=overlay",
+			)),
+		Entry("Azure CNI w Overlay w Cilium",
+			"azure", "overlay", "cilium",
+			"none",
+			sets.New(
+				"kubernetes.azure.com/azure-cni-overlay=true",
+				"kubernetes.azure.com/network-subnet=karpentersub",
+				"kubernetes.azure.com/nodenetwork-vnetguid=test-vnet-guid",
+				"kubernetes.azure.com/podnetwork-type=overlay",
```

Comment:
We could have a separate set as well with these strings that are used in multiple places, and reuse them. Not trying to tackle in this PR, but worthy refactor PR to follow-up.